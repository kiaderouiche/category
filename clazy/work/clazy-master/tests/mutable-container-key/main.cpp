#include <QtCore/QHash>
#include <QtCore/QMap>
#include <QtCore/QPointer>
#include <QtCore/QString>
#include <memory>
struct A {};
typedef QPointer<A> A_QPointer;
using namespace std;
void test()
{
    QPointer<A> pa;
    A a;

    QHash<A, QString> hash1;
    QHash<A*, QString> hash2;
    QHash<QPointer<A>, QString> hash3; // Warning
    QHash<A_QPointer, QString> hash4; // Warning

    QMap<A, QString> map1;
    QMap<A*, QString> map2;
    QMap<QPointer<A>, QString> map3; // Warning
    QMap<A_QPointer, QString> map4; // Warning
    QMap<weak_ptr<A>, QString> map5; // Warning

    QList<QPointer<A>> list;
}
