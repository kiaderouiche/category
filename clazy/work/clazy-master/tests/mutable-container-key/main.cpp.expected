mutable-container-key/main.cpp:16:5: warning: Associative container key might be modified externally [-Wclazy-mutable-container-key]
mutable-container-key/main.cpp:17:5: warning: Associative container key might be modified externally [-Wclazy-mutable-container-key]
mutable-container-key/main.cpp:21:5: warning: Associative container key might be modified externally [-Wclazy-mutable-container-key]
mutable-container-key/main.cpp:22:5: warning: Associative container key might be modified externally [-Wclazy-mutable-container-key]
mutable-container-key/main.cpp:23:5: warning: Associative container key might be modified externally [-Wclazy-mutable-container-key]
